cmd/sink/logs/logs.go

```diff
@@ -5,93 +5,53 @@ package logs
 
 import (
 	"context"
+	"errors"
 	"fmt"
-	"log/slog"
+	"os"
 	"strings"
 
 	ocel "github.com/kubearchive/kubearchive/pkg/cel"
+	"github.com/kubearchive/kubearchive/pkg/files"
 	"github.com/kubearchive/kubearchive/pkg/logurls"
 	"github.com/kubearchive/kubearchive/pkg/models"
-	corev1 "k8s.io/api/core/v1"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
-	"k8s.io/client-go/kubernetes"
 )
 
 const (
 	celPrefix        = "cel:"
 	containerNameCel = "cel:spec.containers.map(m, m.name)"
+	loggingDirEnvVar = "KUBEARCHIVE_LOGGING_DIR"
 )
 
-var (
-	configCmRef = corev1.ObjectReference{
-		Kind:      "ConfigMap",
-		Name:      "kubearchive-config",
-		Namespace: "kubearchive",
-	}
-	loggingCmRef = corev1.ObjectReference{
-		Kind:      "ConfigMap",
-		Namespace: "kubearchive",
-	}
-	loggingCmSelector = corev1.ConfigMapKeySelector{
-		Key: "logging-configmap-name",
+func getKubeArchiveLoggingConfig() (map[string]string, error) {
+	loggingDir, exists := os.LookupEnv(loggingDirEnvVar)
+	if !exists {
+		return nil, errors.New("Environment variable not set: " + loggingDirEnvVar)
 	}
-)
-
-func getKubeArchiveLoggingCm(ctx context.Context, kubeClient kubernetes.Interface) *corev1.ConfigMap {
-	emptyCm := &corev1.ConfigMap{Data: map[string]string{}}
-	kaConfigCm, err := kubeClient.CoreV1().ConfigMaps(configCmRef.Namespace).Get(
-		ctx,
-		configCmRef.Name,
-		metav1.GetOptions{},
-	)
+	configFiles, err := files.FilesInDir(loggingDir)
 	if err != nil {
-		slog.Warn(
-			"error while getting ConfigMap",
-			"ConfigMap",
-			fmt.Sprintf("%s/%s", configCmRef.Namespace, configCmRef.Name),
-			"error",
-			err.Error(),
-		)
-		return emptyCm
+		return nil, fmt.Errorf("Could not read logging config: %w", err)
 	}
-	var exists bool
-	loggingCmRef.Name, exists = kaConfigCm.Data[loggingCmSelector.Key]
-	if !exists {
-		slog.Warn("logging ConfigMap not provided", "key", loggingCmSelector.Key)
-		return emptyCm
-	}
-	loggingCm, err := kubeClient.CoreV1().ConfigMaps(loggingCmRef.Namespace).Get(
-		ctx,
-		loggingCmRef.Name,
-		metav1.GetOptions{},
-	)
+	loggingConf, err := files.LoggingConfigFromFiles(configFiles)
 	if err != nil {
-		slog.Warn(
-			"error while getting ConfigMap",
-			"ConfigMap",
-			fmt.Sprintf("%s/%s", loggingCmRef.Namespace, loggingCmRef.Name),
-			"error",
-			err.Error(),
-		)
-		return emptyCm
+		return nil, fmt.Errorf("Could not get value for logging config: %w", err)
 	}
-	return loggingCm
+	return loggingConf, nil
 }
 
 type UrlBuilder struct {
 	logMap map[string]interface{}
 }
 
-func NewUrlBuilder(ctx context.Context, kubeClient kubernetes.Interface) (*UrlBuilder, error) {
-	if kubeClient == nil {
-		return &UrlBuilder{}, fmt.Errorf("Cannot create UrlBuilder with nil kubernetes client")
+func NewUrlBuilder() (*UrlBuilder, error) {
+	loggingConf, err := getKubeArchiveLoggingConfig()
+	if err != nil {
+		return &UrlBuilder{}, err
```

Comment:
Why return an empty `UrlBuilder`?  This is going to allow calls to `WriteLogs` even when logging isn't configured.  

I think the check to call `Writelogs` should check:
```
if logUrlBuilder != nil &&  strings.ToLower(obj.GetKind()) == "pod" {
```

Reply:
done